airflow/providers/google/cloud/hooks/cloud_run.py

```diff
@@ -0,0 +1,184 @@
+# under the License.
+from __future__ import annotations
+
+import itertools
+import json
+from typing import Iterable, Sequence
+
+from google.api_core import operation
+from google.cloud.run_v2 import (
+    CreateJobRequest,
+    DeleteJobRequest,
+    GetJobRequest,
+    Job,
+    JobsAsyncClient,
+    JobsClient,
+    ListJobsRequest,
+    RunJobRequest,
+    UpdateJobRequest,
+)
+from google.cloud.run_v2.services.jobs import pagers
+from google.longrunning import operations_pb2
+
+from airflow.exceptions import AirflowException
+from airflow.providers.google.common.consts import CLIENT_INFO
+from airflow.providers.google.common.hooks.base_google import PROVIDE_PROJECT_ID, GoogleBaseHook
+
+
+class CloudRunHook(GoogleBaseHook):
+    """
+    Hook for the Google Cloud Run service.
+    :param gcp_conn_id: The connection ID to use when fetching connection info.
+    :param impersonation_chain: Optional service account to impersonate using short-term
+    credentials, or chained list of accounts required to get the access_token
+    of the last account in the list, which will be impersonated in the request.
+    If set as a string, the account must grant the originating account
+    the Service Account Token Creator IAM role.
+    If set as a sequence, the identities from the list must grant
+    Service Account Token Creator IAM role to the directly preceding identity, with first
+    account from the list granting this role to the originating account.
+    """
+
+    def __init__(
+        self,
+        gcp_conn_id: str = "google_cloud_default",
+        impersonation_chain: str | Sequence[str] | None = None,
+        **kwargs,
+    ) -> None:
+        if kwargs.get("delegate_to") is not None:
+            raise RuntimeError(
+                "The `delegate_to` parameter has been deprecated before and finally removed in this version"
+                " of Google Provider. You MUST convert it to `impersonate_chain`"
+            )
+        super().__init__(gcp_conn_id=gcp_conn_id, impersonation_chain=impersonation_chain)
+        self._client: JobsClient | None = None
+
+    def get_conn(self) -> JobsClient:
+        """
+        Retrieves connection to Cloud Run.
+        :return: JobsClient.
+        """
+        if self._client is None:
+            self._client = JobsClient(credentials=self.get_credentials(), client_info=CLIENT_INFO)
+        return self._client
+
+    @GoogleBaseHook.fallback_to_default_project_id
+    def delete_job(self, job_name: str, region: str, project_id: str = PROVIDE_PROJECT_ID) -> Job:
+
+        create_request = DeleteJobRequest()
+        create_request.name = f"projects/{project_id}/locations/{region}/jobs/{job_name}"
+
+        operation = self.get_conn().delete_job(create_request)
+        return operation.result()
+
+    @GoogleBaseHook.fallback_to_default_project_id
+    def create_job(self, job_name: str, job, region: str, project_id: str = PROVIDE_PROJECT_ID) -> Job:
```

Comment:
```suggestion
    def create_job(self, job_name: str, job: Job | dict, region: str, project_id: str = PROVIDE_PROJECT_ID) -> Job:
```

Same in `update_job`